Automattic/wp-calypso
# Newsletter flow: remove intro

We rely on wordpress.com/newsletter as intro instead.



Remove intro step entirely. Alternative to https://github.com/Automattic/wp-calypso/pull/82984

![image](https://github.com/Automattic/wp-calypso/assets/87168/f8f69ede-450f-4985-bc92-6b0ccda4a056)

The support page and other places where we link to intro are enough context/intro, and we want them onto action asap instead of asking again to start. Slack convo (p1697035720638429-slack-C02NQ4HMJKV).

## Proposed Changes

* Remove intro step from newsletter flow, and logic managing conditionally showing it if we came from the marketing page.

## Testing Instructions



* Test urls like
    /setup/newsletter?ref=newsletter-lp
    /setup/newsletter?ref=support-site-inline
    /setup/newsletter/intro?ref=newsletter-lp
    /setup/newsletter/intro?ref=support-site-inline
* Test logged out, and logged in
* Test that you see "signup start" event and normal "step start" events

## Pre-merge Checklist
